pkg/virt-api/webhooks/validating-webhook/admitters/vms-delete-admitter.go

```diff
@@ -0,0 +1,92 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package admitters
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+	webhookutils "kubevirt.io/kubevirt/pkg/util/webhooks"
+	"kubevirt.io/kubevirt/pkg/virt-api/webhooks"
+)
+
+type VMsDeleteAdmitter struct {
+	VirtClient kubecli.KubevirtClient
+}
+
+func NewVMsDeletionAdmitter(client kubecli.KubevirtClient) *VMsDeleteAdmitter {
+	return &VMsDeleteAdmitter{
+		VirtClient: client,
+	}
+}
+
+func (admitter *VMsDeleteAdmitter) Admit(ctx context.Context, ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
+	if !webhookutils.ValidateRequestResource(ar.Request.Resource, webhooks.VirtualMachineGroupVersionResource.Group, webhooks.VirtualMachineGroupVersionResource.Resource) {
+		err := fmt.Errorf("expected resource to be '%s'", webhooks.VirtualMachineGroupVersionResource.Resource)
+		return webhookutils.ToAdmissionResponseError(err)
+	}
+
+	if ar.Request.DryRun != nil && *ar.Request.DryRun {
+		log.Log.V(3).Infof("Skipping patch for VMI '%s/%s' due to dry-run", ar.Request.Namespace, ar.Request.Name)
+		return &admissionv1.AdmissionResponse{Allowed: true}
+	}
+
+	if ar.Request.Options.Raw != nil {
+		deleteOpts := &metav1.DeleteOptions{}
+		if err := json.Unmarshal(ar.Request.Options.Raw, deleteOpts); err != nil {
+			log.Log.Errorf("Failed to unmarshal DeleteOptions for VM '%s/%s': %v", ar.Request.Namespace, ar.Request.Name, err)
+			return &admissionv1.AdmissionResponse{Allowed: true}
+		}
+
+		if deleteOpts.GracePeriodSeconds != nil {
+			grace := *deleteOpts.GracePeriodSeconds
+
+			gracePatch, err := patch.New(patch.WithReplace("/spec/terminationGracePeriodSeconds", grace)).GeneratePayload()
+			if err != nil {
+				log.Log.Errorf("Failed to generate JSON patch for grace period: %v", err)
+			} else {
+				_, patchErr := admitter.VirtClient.VirtualMachineInstance(ar.Request.Namespace).Patch(
+					ctx,
+					ar.Request.Name,
+					types.JSONPatchType,
+					gracePatch,
+					metav1.PatchOptions{},
+				)
+				if patchErr != nil && !errors.IsNotFound(patchErr) {
+					log.Log.Errorf("Failed to patch VMI terminationGracePeriodSeconds for VM '%s/%s': %v", ar.Request.Namespace, ar.Request.Name, patchErr)
+				}
+			}
+		}
+	}
+
+	return &admissionv1.AdmissionResponse{
```

Comment:
supernit - inline as above

Reply:
I felt that having the return in both the if and else blocks made the code a bit more cluttered. Leaving it as a single return at the end seemed cleaner and easier to follow. Happy to change it if there’s a strong preference though.

Reply:
```suggestion
	return &admissionv1.AdmissionResponse{Allowed: true}
```

Sorry I mean this, just a style supernit.